### ISSUE TITLE
InfluxDB 1.2.2 fails to ingest collectd-native data using [[collectd]]

### ISSUE THREAD

**Issue by Here-Be-Dragons** [influxdata/influxdb#8184]
### Bug report

__System info:__ [Include InfluxDB version, operating system name, and other relevant details]
InfluxDB version: 1.2.2 (influxdb-1.2.2.x86_64)
OS: RHEL7
Collectd version: collectd-5.5.0 (collectd-5.5.0-3.el7.x86_64)
__Steps to reproduce:__

1. Install Collectd and InfluxDB to the versions listed above.
2. Enable `LoadPlugin network` in collectd with typical configurations to send to InfluxDB in `collectd.conf`:
   ```xml
   <Plugin network>
      <Server "localhost" "25826">
        SecurityLevel "None"
     </Server>
     ReportStats true
     MaxPacketSize 1452
   </Plugin>
   ```
3. Configure InfluxDB to receive data in Collectd format in `influxdb.conf`:
   ```bash
   [[collectd]]
     enabled = true
     bind-address = ":25826"
     database = "collectd"
     retention-policy = ""
     typesdb = "/usr/share/collectd/types.db"
     security-level = "none"
     auth-file = "/etc/collectd/auth_file"
     batch-size = 5000
     batch-pending = 10
     batch-timeout = "10s"
     read-buffer = 0
   ```
4. use `curl_xml` or `snmp` plugins in collectd (possibly others)  
__Expected behavior:__ [What you expected to happen]
InfluxDB stores data sent by Collectd in the specified table

__Actual behavior:__ [What actually happened]
No data is stored, and the following error message is printed to `/var/log/messages` any time collectd sends data:  
`<timestamp> <hostname> influxd: [I] 2017-03-22T07:24:47Z Collectd parse error: invalid data service=collectd`

__Additional info:__ [Include gist of relevant config, logs, etc.]
Downgrading to InfluxDB 1.1.1 resolves this issue and is not affected by the above bug.

**Comment by eighteen14k** [comment]
I've seen this error with influxdb 1.2.2 and resolved it ... what did I do ... forget :(

Just out of curiosity, if you drop the "MaxPacketSize" down to 1400, does the problem go away?

**Comment by kozhevn** [comment]
Probably, it may be fixed the same way as described in #8140
The file 'types.db' is missing in .deb package and it should be located manually at /usr/local/share/collectd
It may be taken here: https://raw.githubusercontent.com/collectd/collectd/master/src/types.db

**Comment by colemanfuel** [comment]
I had a similar issue on CentOS 7 with influxdb-1.5.2-1.x86_64. My 'types.db' existed and was accurate.

Downgrading to influxdb-1.1.1-1.x86_64 resolved this for me.
